src/types/resources/itemImageData.d.ts

```diff
@@ -14,6 +14,7 @@ declare module 'resources' {
         imageHash: string;
         originalMime: string;
         originalName: string;
+        itemImageId: number;
```

Comment:
is this used for something?
since ItemImageData has a relation to ItemImage, rather add that

Reply:
The itemImageData object only seems to populate the ItemImage relation when it fetches data `withRelated=true`. Which is not necessarily always the case. Also, in the context that it was being used, it seemed unnecessary to fetch the entire ItemImage just to extract the relation ID. And I couldn't get a reference to the related ItemImage ID without adding the property explicitly.

Is there another way I could have done this (with ItemImageData.ItemImage in the case it was used in being undefined)?

Reply:
Propably not.

I dont mind itemImageId being there, or it being used, but ItemImage should be added there too, since the relation exists. I've been adding the related models to interfaces usually when I need them and the relation id fields only if absolutely necessary. This is because I value code readability rather than efficiency, but also because fetching the whole model with relations usually leads to cleaner and shorter code. The withRelated is there for the efficiency reason when thats needed. If I want to use a related models id, I'll usually do it explicitly using Model.RelatedModel.id, because imo that's an exact way of describing to the reader exactly what value I am using. If I point to the value using Model.relatedModelId, I'm telling the reader that I'm just using some value from the model.